src/lpf2hub.ts

```diff
@@ -608,5 +539,187 @@ export class LPF2Hub extends Hub {
+             * @event LPF2Hub#color
+             * @param {string} port
+             * @param {Color} color
+             */
+            this.emit("color", port.id, color);
+        }
+
+        if (distance !== undefined) {
+            /**
+             * Emits when a distance sensor is activated.
+             * @event LPF2Hub#distance
+             * @param {string} port
+             * @param {number} distance Distance, in millimeters.
+             */
+            this.emit("distance", port.id, distance);
+        }
+
+        if (color !== undefined && distance !== undefined) {
+            /**
+             * A combined color and distance event, emits when the sensor is activated.
+             * @event LPF2Hub#colorAndDistance
+             * @param {string} port
+             * @param {Color} color
+             * @param {number} distance Distance, in millimeters.
+             */
+            this.emit("colorAndDistance", port.id, color, distance);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.COUNT) {
+            /**
+             * Emits when a count mode distance sensor is activated.
+             * @event LPF2Hub#count
+             * @param {string} port
+             * @param {number} count
+             */
+            this.emit("count", port.id, data[4]);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.REFLT) {
+            /**
+             * Emits when a reflect mode color sensor is activated.
+             * @event LPF2Hub#reflectivity
+             * @param {string} port
+             * @param {number} reflectivity
+             */
+            this.emit("reflectivity", port.id, data[4]);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.AMBI) {
+            /**
+             * Emits when a ambiant mode color sensor is activated.
+             * @event LPF2Hub#luminosity
+             * @param {string} port
+             * @param {number} luminosity
+             */
+            this.emit("luminosity", port.id, data[4]);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.RGB_I) {
+            /**
+             * Emits when a RGB mode color sensor is activated.
+             * @event LPF2Hub#rgb
+             * @param {string} port
+             * @param {number} r
+             * @param {number} g
+             * @param {number} b
+             */
+            this.emit("rgb", port.id, data[4], data[6], data[8]);
+        }
+    }
+
+    private _parseBoostTilt(port: Port, data: Buffer) {
+        const values: number[] = [];
+        let event = '';
+
+        switch (port.mode) {
+            case Consts.BoostTiltModes.ANGLE: {
+                values.push(data.readInt8(4));
+                values.push(data.readInt8(5));
+                event = 'angle';
+                break;
+            }
+
+            case Consts.BoostTiltModes.TILT: {
+                values.push(data[4]);
+                event = 'tilt';
```

Comment:
In this mode the sensor returns a discrete direction of tilt (1, 3, 5, or 7) for each major direction as far as I remember. This is not a very useful mode IMO, but previously the 'tilt' event was reporting acceleration (boost and control+ hubs) or angles (wedo sensor), so this is another change to the API.

Reply:
I have to admit that I don't know either why this mode exists as it could be calculated from angle...

Maybe to save some power and require angle only when you have a tilt trigger ?